spec/scnnr/client_spec.rb

```diff
@@ -47,13 +95,18 @@
     let(:options) { {} }
     let(:expected_recognition) { Scnnr::Recognition.new }
 
+    before do
+      allow(Scnnr::Connection).to receive(:new).with(uri, :post, api_key, logger) { mock_connection }
```

Comment:
Why did you change `expect` -> `allow` around `Connection` and `Response` methods?
I think these methods should be ensured to be called :)

Reply:
Oh, good point - I'm not sure if `receive(:new)` is something that needs to be ensured, but `receive(:send_stream)` definitely is! I just got carried away with the `before/allow` vs `it/expect` approach ...

Is this a case where you would just use `expect` in a before block because it's setup as well as a check, or do you have another idea of how to arrange this?

Reply:
As you said, `receive(:new)` may not have to be ensured. 
In my opinion, I want to ensure that one request is created by `Connection` and one recognition is created by `Response` :)

So, for example, in the `recognize_image` spec, is it better to do as folloing? 

```
describe '#recognize_image' do
  subject { client.recognize_image(image, options) }

  let(:image) { fixture('images/sample.png') }
  let(:uri) { client.send(:construct_uri, 'recognitions', options) }
  let(:options) { {} }
  let(:expected_recognition) { Scnnr::Recognition.new }

  before do
    allow(Scnnr::Connection).to receive(:new).with(uri, :post, api_key, logger) { mock_connection }
    allow(Scnnr::Response).to receive(:new).with(mock_origin_response, boolean) { mock_response }
  end

  it do
    expect(mock_connection).to receive(:send_stream).with(image) { mock_origin_response }
    expect(mock_response).to receive(:build_recognition) { expected_recognition }
    expect(subject).to eq expected_recognition
  end

  it_behaves_like 'posting an image'
end
```

What do you think?

Reply:
Actually, using `PollingManager.start` solved the problem anyway, since I can just check that method is called and test the timeout behaviour more simply in the PollingManager specs. I think that proves the factoring is better!

For your interest, before moving those specs I tried duplicating those lines as `expect`s in the first spec and `allow`s in the examples block, like this:

```ruby
  it do
    expect(Scnnr::Connection).to receive(:new).with(uri, :post, api_key, logger) { mock_connection }
    expect(mock_connection).to receive(:send_stream).with(image) { mock_origin_response }
    expect(Scnnr::Response).to receive(:new).with(mock_origin_response, boolean) { mock_response }
    expect(mock_response).to receive(:build_recognition) { expected_recognition }
    expect(subject).to eq expected_recognition
  end

  context 'handling timeouts' do
    before do
        allow(Scnnr::Connection).to receive(:new).with(uri, :post, api_key, logger) { mock_connection }
        allow(mock_connection).to receive(:send_stream).with(image) { mock_origin_response }
        allow(Scnnr::Response).to receive(:new).with(mock_origin_response, boolean) { mock_response }
    end

    include_examples 'posting an image'
  end
```

I think that would express more clearly what the different tests do - the first one checks the core behaviour of the method, and the other ones don't care about any behaviour except the timeouts.

Anyway, it's much nicer having the specs in PollingManager.